src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,46 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> element;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+
+        public Node(T item, Node<T> next) {
+            this.item = item;
+            this.next = next;
+        }
+    }
 
+    public void push(T value) {
+        element = new Node<>(value, element);
+        size++;
     }
 
     public T peek() {
-        return null;
+        isEmpty();
+        return element.item;
     }
 
     public T pop() {
-        return null;
+        isEmpty();
+        final T result = element.item;
+        element = element.next;
+        size--;
+        return result;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    public void isEmpty() {
```

Comment:
method with such naming usually returns boolean
```suggestion
    private void validateSize() {
```